### PR TITLE
dra: resourceclass missing

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

When filtering fails because a ResourceClass is missing, we can treat the pod
as "unschedulable" as long as we then also register a cluster event that wakes
 up the pod. This is more efficient than periodically retrying.

#### Special notes for your reviewer:

Includes one other drive-by fix for event registration.

#### Does this PR introduce a user-facing change?
```release-note
scheduler: handling of unschedulable pods because a ResourceClass is missing is a bit more efficient and no longer relies on periodic retries
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/3063
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go [hunk 1]
```diff
@@ -272,6 +272,8 @@ func (pl *dynamicResources) EventsToRegister() []framework.ClusterEventWithHint
 		// A resource might depend on node labels for topology filtering.
 		// A new or updated node may make pods schedulable.
 		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.UpdateNodeLabel}},
+		// A pod might be waiting for a class to get created or modified.
+		{Event: framework.ClusterEvent{Resource: framework.ResourceClass, ActionType: framework.Add | framework.Update}},
```
Comment (sanposhiho) [inline pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go:276]: In which scenario rejected Pods should be moved from unsched to backoffQ/activeQ via update events?

Comment (pohly) [reply pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go:276]: The node filter in the class might change. I have a hunch that this then leads to an unschedulable pod (no suitable nodes at all) which might be stuck without this update event, already now (i.e. a bug).

Let me write a test for this scenario.

Comment (pohly) [reply pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go:276]: It's as I suspected: the [new test](https://github.com/kubernetes/kubernetes/compare/131c81882a07f3c477e8dd580ad6b14ae634665d..a479431782b89c194335e5f61f89147349f221dd) hangs when the `Update` event is not registered.

**Comment by pohly** [comment]
@sanposhiho: can you take another look? The PR should be complete now and I fixed the gofmt issue.

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve
/retest

**Comment by pacoxu** [comment]
/retest
oomkiller failure is a top flake

**Pull request merged**